codec/decoder/plus/src/welsDecoderExt.cpp

```diff
@@ -699,8 +699,15 @@ DECODING_STATE CWelsDecoder::DecodeFrameNoDelay (const unsigned char* kpSrc,
     if (m_sReoderingStatus.iNumOfPicts) {
       WAIT_EVENT (&m_sBufferingEvent, WELS_DEC_THREAD_WAIT_INFINITE);
       RESET_EVENT (&m_sReleaseBufferEvent);
-      ReleaseBufferedReadyPicture (NULL, ppDst, pDstInfo);
-      SET_EVENT (&m_sReleaseBufferEvent);
+      if (!m_sReoderingStatus.bHasBSlice) {
+        if (m_sReoderingStatus.iNumOfPicts > 1) {
```

Comment:
if m_sReoderingStatus.iNumOfPicts <= 1 there is no call to either ReleaseBufferedReadyPictureNoReorder() nor ReleaseBufferedReadyPictureReorder()
Is that intended?

Reply:
Yes. I think it is intended. But I'll double check it and let you know.

Reply:
@joakim-tjernlund You are right. line 703 is better with  if (m_sReoderingStatus.iNumOfPicts >= 1) {. But I guess it is harmless with if (m_sReoderingStatus.iNumOfPicts > 1) {.

Reply:
You sure it is harmless? Seem odd to me to skip calling any of ReleaseBufferedReadyPictureNoReorder()/ReleaseBufferedReadyPictureReorder() if m_sReoderingStatus.iNumOfPicts == 1